apps/geo/serializers.py

```diff
@@ -148,3 +151,69 @@ class Meta:
             'region_title', 'admin_level_level', 'admin_level_title',
             'parent'
         )
+
+
+class RegionGqSerializer(UserResourceSerializer):
+    project = serializers.PrimaryKeyRelatedField(queryset=Project.objects.all())
+    client_id = serializers.CharField(required=False)
+
+    class Meta:
+        model = Region
+        fields = ['title', 'code', 'project', 'client_id']
+
+    def validate_project(self, project):
+        if not project.can_modify(self.context['request'].user):
+            raise serializers.ValidationError('Permission Denied')
+        return project
+
+    def validate(self, data):
+        if self.instance and self.instance.is_published:
+            raise serializers.ValidationError('Published region can\'t be changed. Please contact Admin')
+        return data
+
+    def create(self, validated_data):
+        project = validated_data.pop('project', None)
+        region = super().create(validated_data)
+        project.regions.add(region)
+        return region
+
+
+class AdminLevelGqlSerializer(UserResourceSerializer):
+    region = serializers.PrimaryKeyRelatedField(queryset=Region.objects.all())
+
+    class Meta:
+        model = AdminLevel
+        exclude = ('geo_area_titles',)
+
+    def validate_region(self, region):
+        if not region.can_modify(self.context['request'].user):
+            raise serializers.ValidationError('Invalid Region')
+        if region.is_published:
+            raise serializers.ValidationError('The AdminLevel cannot be altered once the region has been published.')
+        return region
+
+    def create(self, validated_data):
+        admin_level = super().create(validated_data)
+
+        if not settings.TESTING:
+            transaction.on_commit(lambda: load_geo_areas.delay(admin_level.region.id))
```

Comment:
We don't need this anymore. 

```suggestion
        transaction.on_commit(lambda: load_geo_areas.delay(admin_level.region.id))
```